tests/test_subannual.R

```diff
@@ -0,0 +1,293 @@
+library(testthat)
+library(tsbox)
+
+t3 <- ts_rbind(ts_dt(mdeaths), AirPassengers)   ##### this function contains error, should have both
+# AirPassengers and mdeaths in column "var"
+t4 <- ts_rbind(ts_xts(AirPassengers), ts_tbl(mdeaths)) ##### this function contains error, should have both
+# AirPassengers and mdeaths in column "var"
+
+
+context("irregular conversion handling")
+
+####### Monthly series: AirPassengers #####
+test_that("conversion produces right classes", {
+  expect_s3_class(ts_xts(window(AirPassengers, start = c(1949, 5))), "xts")
+  expect_s3_class(ts_ts(window(AirPassengers, start = c(1949, 5))), "ts")
+  expect_s3_class(ts_df(window(AirPassengers, start = c(1949, 5))), "data.frame")
+  expect_s3_class(ts_dt(window(AirPassengers, start = c(1949, 5))), "data.table")
+  expect_s3_class(ts_tbl(window(AirPassengers, start = c(1949, 5))), "tbl_df")
+  
+  expect_s3_class(ts_xts(ts_xts(window(AirPassengers, start = c(1949, 5)))), "xts")
+  expect_s3_class(ts_ts(ts_xts(window(AirPassengers, start = c(1949, 5)))), "ts")
+  expect_s3_class(ts_df(ts_xts(window(AirPassengers, start = c(1949, 5)))), "data.frame")
+  expect_s3_class(ts_dt(ts_xts(window(AirPassengers, start = c(1949, 5)))), "data.table")
+  expect_s3_class(ts_tbl(ts_xts(window(AirPassengers, start = c(1949, 5)))), "tbl_df")
+  
+  expect_s3_class(ts_xts(ts_df(window(AirPassengers, start = c(1949, 5)))), "xts")
+  expect_s3_class(ts_ts(ts_df(window(AirPassengers, start = c(1949, 5)))), "ts")
+  expect_s3_class(ts_df(ts_df(window(AirPassengers, start = c(1949, 5)))), "data.frame")
+  expect_s3_class(ts_dt(ts_df(window(AirPassengers, start = c(1949, 5)))), "data.table")
+  expect_s3_class(ts_tbl(ts_df(window(AirPassengers, start = c(1949, 5)))), "tbl_df")
+  
+  expect_s3_class(ts_xts(ts_dt(window(AirPassengers, start = c(1949, 5)))), "xts")
+  expect_s3_class(ts_ts(ts_dt(window(AirPassengers, start = c(1949, 5)))), "ts")
+  expect_s3_class(ts_df(ts_dt(window(AirPassengers, start = c(1949, 5)))), "data.frame")
+  expect_s3_class(ts_dt(ts_dt(window(AirPassengers, start = c(1949, 5)))), "data.table")
+  expect_s3_class(ts_tbl(ts_dt(window(AirPassengers, start = c(1949, 5)))), "tbl_df")
+  
+  expect_s3_class(ts_xts(ts_tbl(window(AirPassengers, start = c(1949, 5)))), "xts")
+  expect_s3_class(ts_ts(ts_tbl(window(AirPassengers, start = c(1949, 5)))), "ts")
+  expect_s3_class(ts_df(ts_tbl(window(AirPassengers, start = c(1949, 5)))), "data.frame")
+  expect_s3_class(ts_dt(ts_tbl(window(AirPassengers, start = c(1949, 5)))), "data.table")
+  expect_s3_class(ts_tbl(ts_tbl(window(AirPassengers, start = c(1949, 5)))), "tbl_df")
+  
+})
+
+test_that("conversion between objects works as expected: AirPassengers", {
+  
+  x.ts <- window(AirPassengers, start = c(1949, 5))
+  
+  x.xts <- ts_xts(x.ts, cname = 'AirPassengers')
```

Comment:
Was this a workaround, or have I done something similar? If you did it, can you comment why you did it?

Reply:
I'am not clear why you said that this is a workaround.

Reply:
I wonder where the `cname` argument comes from? It is not a standard argument. Have you figured that out, or have I done something similar somewhere else?

Reply:
I saw the cname in the test_object_conversion.R:

  expect_equal(ts_xts(ts_ts(x.xts), cname = "discoveries"), x.xts)
  expect_equal(ts_xts(ts_df(x.xts)), x.xts)
  expect_equal(ts_xts(ts_dt(x.xts)), x.xts)
  expect_equal(ts_xts(ts_tbl(x.xts)), x.xts)

Reply:
Ah, wasn’t aware of that. Thanks